native/dggsvp3.go

```diff
@@ -260,8 +260,8 @@ func (impl Implementation) Dggsvp3(jobU, jobV, jobQ lapack.GSVDJob, m, p, n int,
 		}
 
 		// Clean up A.
-		for i := k + 1; i < k+l; i++ {
-			r := a[i*lda+n-l : i*lda+i-k+n-l]
+		for i := k + 1; i < m; i++ {
+			r := a[i*lda+n-l : i*lda+min(i-k+n-l, n)]
```

Comment:
Yes, I noticed that before, and then forgot when I was fixing this. I'll do that now.